src/json/serialization.cr

```diff
@@ -419,8 +430,20 @@ module JSON
 
         case discriminator_value
         {% for key, value in mapping %}
-          when {{key.id.stringify}}
-            {{value.id}}.from_json(json)
+          {% if mapping.is_a?(NamedTupleLiteral) %}
+            when {{key.id.stringify}}
+          {% else %}
+            {% if key.is_a?(StringLiteral) %}
+              when {{key.id.stringify}}
```

Comment:
I think `key.id.stringify` is just `key`?

Reply:
Oh, you're right! I just kept it this way because I was looking at the `NamedTupleLiteral` stringification.